core/src/test/scala/org/locationtech/rasterframes/functions/FocalFunctionsSpec.scala

```diff
@@ -0,0 +1,346 @@
+/*
+ * This software is licensed under the Apache 2 license, quoted below.
+ *
+ * Copyright 2020 Astraea, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *     [http://www.apache.org/licenses/LICENSE-2.0]
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.locationtech.rasterframes.functions
+
+import geotrellis.raster.mapalgebra.focal.{Circle, Kernel, Square}
+import geotrellis.raster.{BufferTile, CellSize}
+import geotrellis.raster.testkit.RasterMatchers
+import org.locationtech.rasterframes.ref.{RFRasterSource, RasterRef, Subgrid}
+import org.locationtech.rasterframes.tiles.ProjectedRasterTile
+import org.locationtech.rasterframes._
+import geotrellis.raster.Tile
+import geotrellis.raster.mapalgebra.local.Implicits._
+import org.locationtech.rasterframes.encoders.serialized_literal
+
+import java.nio.file.Paths
+
+class FocalFunctionsSpec extends TestEnvironment with RasterMatchers {
+
+  import spark.implicits._
+
+  describe("focal operations") {
+    lazy val path =
+      if(Paths.get("").toUri.toString.endsWith("core/")) Paths.get("src/test/resources/L8-B7-Elkton-VA.tiff").toUri
```

Comment:
Magical dragons?

Reply:
@metasim 🤦 I didn't figure out how to make it better.

Reply:
Clever hack!